virtcontainers/acrn_arch_base.go

```diff
@@ -0,0 +1,762 @@
+	}
+}
+
+func (config *Config) appendMemory() {
+	if config.Memory.Size != "" {
+		config.acrnParams = append(config.acrnParams, "-m")
+		config.acrnParams = append(config.acrnParams, config.Memory.Size)
+	}
+}
+
+func (config *Config) appendCPUs() {
+	if config.NumCPU != 0 {
+		config.acrnParams = append(config.acrnParams, "-c")
+		config.acrnParams = append(config.acrnParams, fmt.Sprintf("%d", config.NumCPU))
+	}
+
+}
+
+func (config *Config) appendKernel() {
+	if config.Kernel.Path != "" {
+		config.acrnParams = append(config.acrnParams, "-k")
+		config.acrnParams = append(config.acrnParams, config.Kernel.Path)
+
+		if config.Kernel.Params != "" {
+			config.acrnParams = append(config.acrnParams, "-B")
+			config.acrnParams = append(config.acrnParams, config.Kernel.Params)
+		}
+	}
+}
+
+// LaunchAcrn can be used to launch a new acrn instance.
+//
+// The Config parameter contains a set of acrn parameters and settings.
+//
+// This function writes its log output via logger parameter.
+func LaunchAcrn(config Config, logger *logrus.Entry) (int, string, error) {
+
+	baselogger = logger
+	config.appendUUID()
+	config.appendACPI()
+	config.appendMemory()
+	config.appendCPUs()
+	config.appendDevices()
+	config.appendKernel()
+	config.appendName()
+
+	return LaunchCustomAcrn(context.Background(), config.Path, config.acrnParams, logger)
+}
+
+// LaunchCustomAcrn can be used to launch a new acrn instance.
+//
+// The path parameter is used to pass the acrn executable path.
+//
+// params is a slice of options to pass to acrn-dm
+//
+// This function writes its log output via logger parameter.
+func LaunchCustomAcrn(ctx context.Context, path string, params []string,
+	logger *logrus.Entry) (int, string, error) {
+
+	errStr := ""
+
+	if path == "" {
+		path = "acrn-dm"
+	}
+
+	/* #nosec */
+	cmd := exec.CommandContext(ctx, path, params...)
+
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+	logger.Infof("launching %s with: %v", path, params)
+
+	err := cmd.Start()
+	if err != nil {
+		logger.Errorf("Unable to launch %s: %v", path, err)
+		errStr = stderr.String()
+		logger.Errorf("%s", errStr)
+	}
+	return cmd.Process.Pid, errStr, err
+}
+
+func (a *acrnArchBase) appendImage(devices []Device, path string) ([]Device, error) {
+	imageFile, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = imageFile.Close() }()
+
+	ImgBlkdevice := BlockDevice{
+		FilePath: path,
+		Index:    0,
+	}
+
+	devices = append(devices, ImgBlkdevice)
+
+	return devices, nil
+}
+
+// appendBridges appends to devices the given bridges
+func (a *acrnArchBase) appendBridges(devices []Device) []Device {
+
+	devices = append(devices,
+		BridgeDevice{
+			Function: 0,
+			Emul:     "hostbridge",
+			Config:   "",
+		},
+	)
+
+	return devices
+}
+
+// appendBridges appends to devices the given bridges
+func (a *acrnArchBase) appendLPC(devices []Device) []Device {
+
+	devices = append(devices,
+		LPCDevice{
+			Function: 0,
+			Emul:     "lpc",
```

Comment:
You've used this string already above so I'd consider creating a `const` for it "just in case" it ever needs to change.

Reply:
added const for "lpc". But this will not be changed.